### PR TITLE
Translate engineconn-plugins-elasticsearch  service classes from Scala to Java 

### PR DESCRIPTION
**Pull request by ChengJie1053**
Translate engineconn-plugins-elasticsearch  service classes from Scala to Java

### CODE REVIEW

File: linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        ResultSetWriter<? extends MetaData, ? extends Record> resultSetWriter =
+            engineExecutorContext.createResultSetWriter(ResultSetFactory.TABLE_TYPE);
+        resultSetWriter.addMetaData(metaData);
+        Arrays.asList(tableResponse.records())
+            .forEach(
+                record -> {
+                  try {
+                    resultSetWriter.addRecord(record);
+                  } catch (IOException e) {
+                    logger.warn("es addRecord failed", e);
```
Comment (peacewong) [inline linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java:143]: If there is an exception, an ErrorExecuteResponse needs to be returned

File: linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        String output = resultSetWriter.toString();
+        IOUtils.closeQuietly(resultSetWriter);
+        return new AliasOutputExecuteResponse(null, output);
+      } else if (elasticSearchResponse instanceof ElasticSearchErrorResponse) {
+        ElasticSearchErrorResponse errorResponse =
+            (ElasticSearchErrorResponse) elasticSearchResponse;
+        return new ErrorExecuteResponse(errorResponse.message(), errorResponse.cause());
+      }
+    } catch (IOException e) {
+      logger.warn("es addMetaData failed", e);
```
Comment (peacewong) [inline linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java:172]: If there is an exception, an ErrorExecuteResponse needs to be returned

File: linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    }
+
+    return null;
```
Comment (peacewong) [inline linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java:175]: An ErrorExecuteResponse needs to be returned

File: linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  }
+
+  private Map<String, String> buildRuntimeParams(EngineConnTask engineConnTask) {
+    // Parameters specified at runtime
+    Map<String, String> executorProperties =
+        engineConnTask.getProperties().entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey, entry -> Objects.toString(entry.getValue(), null)));
+    if (executorProperties == null) {
```
Comment (peacewong) [inline linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java:185]: should first judge the null

Comment (ChengJie1053) [reply linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/ElasticSearchEngineConnExecutor.java:185]: Ok, thanks for reviewing the code

File: linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/client/impl/ElasticSearchExecutorImpl.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        break;
+    }
+  }
+
+  @Override
+  public ElasticSearchResponse executeLine(String code) {
+    String realCode = code.trim();
+    logger.info("es client begins to run {} code:\n {}", runType, realCode.trim());
+    CountDownLatch countDown = new CountDownLatch(1);
+    ElasticSearchResponse[] executeResponse = {
```
[the remark this answers is not in the document]
Comment (ChengJie1053) [reply linkis-engineconn-plugins/elasticsearch/src/main/java/org/apache/linkis/engineplugin/elasticsearch/executor/client/impl/ElasticSearchExecutorImpl.java:75]: image

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**